ginga/AstroImage.py

```diff
@@ -221,27 +184,19 @@ def set_io(self, io):
     def get_data_size(self):
         return self.get_size()
 
-    def get_header(self, create=True, include_primary_header=None):
+    def get_header(self, create=True, include_primary_header=False):
+
         try:
             # By convention, the fits header is stored in a dictionary
             # under the metadata keyword 'header'
             hdr = self.metadata['header']
 
-            if include_primary_header is None:
-                include_primary_header = self.inherit_primary_header
-
-            if include_primary_header and self._primary_hdr is not None:
+            primary_hdr = self.metadata.get('primary_header', None)
+            if include_primary_header and primary_hdr is not None:
                 # Inherit PRIMARY header for display but keep metadata intact
                 displayhdr = AstroHeader()
-                for key in hdr.keyorder:
-                    card = hdr.get_card(key)
-                    bnch = displayhdr.__setitem__(card.key, card.value)
-                    bnch.comment = card.comment
-                for key in self._primary_hdr.keyorder:
-                    if key not in hdr:
-                        card = self._primary_hdr.get_card(key)
-                        bnch = displayhdr.__setitem__(card.key, card.value)
-                        bnch.comment = card.comment
+                displayhdr.merge(hdr)
+                displayhdr.merge(primary_hdr)
```

Comment:
Better be explicit here instead of relying on default setting, which might change in the future. One should never overwrite keyword in data extension from primary header.

```suggestion
                displayhdr.merge(hdr)
                displayhdr.merge(primary_hdr, override_keywords=False)
```